### PR TITLE
support bgpstream broker version 2

### PR DESCRIPTION
**Pull request by digizeph**

### CODE REVIEW

File: lib/bgpstream_record.h [hunk 1]
```diff
@@ -75,6 +75,20 @@ typedef enum {
 
 } bgpstream_record_type_t;
 
+/** The type of livestream */
```
Comment (alistairking) [inline lib/bgpstream_record.h]: i think this can be removed

File: lib/formats/bs_format_rislive.c [hunk 2]
```diff
@@ -610,45 +627,18 @@ int bs_format_rislive_get_next_elem(bgpstream_format_t *format,
...
-                    RDATA->status_msg_state);
-      break;
-    }
+    RDATA->elem->old_state = BGPSTREAM_ELEM_PEERSTATE_UNKNOWN;
+    RDATA->elem->new_state = RDATA->status_msg_state;
     RDATA->end_of_elems = 1;
     rc = 1;
     break;
   case RISLIVE_MSG_TYPE_OPEN:
+    bgpstream_log(BGPSTREAM_LOG_WARN, "RIS Live OPEN support coming soon...");
```
Comment (alistairking) [inline lib/formats/bs_format_rislive.c]: how soon? before v2 release?
if not, i think we should remove this warning and we'll just have to document that we don't support it.

File: lib/transports/bs_transport_cache.c [hunk 3]
```diff
@@ -238,6 +238,7 @@ int bs_transport_cache_create(bgpstream_transport_t *transport)
     }
   }
 
+
```
Comment (alistairking) [inline lib/transports/bs_transport_cache.c]: remove this line

File: tools/bgpreader.c [hunk 4]
```diff
@@ -602,10 +602,14 @@ int main(int argc, char *argv[])
 
   if (interval_start == 0 && !intervalstring) {
     if (di_id == BGPSTREAM_DATA_INTERFACE_BROKER) {
-      fprintf(stderr,
-              "ERROR: At least one time window must be set when using the "
-              "broker data interface\n");
-      goto done;
+      fprintf(stderr, "WARN: No time window specified, defaulting to now to forever interval\n");
```
Comment (alistairking) [inline tools/bgpreader.c]: -> "WARN: No time window specified, defaulting to live mode."

File: tools/bgpreader.c [hunk 5]
```diff
@@ -602,10 +602,14 @@ int main(int argc, char *argv[])
...
+      interval_start = (uint32_t)time(NULL);
```
Comment (alistairking) [inline tools/bgpreader.c]: `epoch_sec()`

File: lib/formats/bs_format_rislive.c (hunk 2 above)
Comment (digizeph) [reply lib/formats/bs_format_rislive.c]: i think it requires quite some efforts and i vote to put this into v2.1 milestone.

**Pull request merged**